### PR TITLE
add all 6 sequence diagrams

### PR DESCRIPTION
**Pull request by TheSpaceCuber**
https://towardsdatascience.com/drawing-a-uml-diagram-in-the-vs-code-53c2e67deffe

Install vscode extension, that's all.

Refer to google docs for actual diagrams.

### CODE REVIEW

**Review by ongweisheng — APPROVED** [review]
LGTM!

File: docs/sequence-diagrams/user-service.puml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@startuml
+Client -> Frontend: Register as new user
+Frontend -> User: POST /api/user/signup
+User -> Frontend: Success
+Frontend -> Client: Redirect to login
+Client -> Frontend: Login
+Frontend -> User: POST /api/user/login
```
Comment (hiongkaihan) [inline docs/sequence-diagrams/user-service.puml:7]: Update route from 
/api/user/signup to /signup
and 
/api/user/login to /login

**Review by hiongkaihan — CHANGES_REQUESTED** [review]

File: docs/sequence-diagrams/user-service.puml (hunk 1 above)
Comment (TheSpaceCuber) [reply docs/sequence-diagrams/user-service.puml]: Okay

**Comment by TheSpaceCuber** [comment]
@hiongkaihan can merge once you approve, any future changes I will make new branch / PR

**Pull request merged**